lib/ui/widgets/patchesSelectorView/patch_options_fields.dart

```diff
@@ -524,29 +528,53 @@ class _TextFieldForPatchOptionState extends State<TextFieldForPatchOption> {
                   ];
                 },
                 onSelected: (String selection) async {
-                  switch (selection) {
-                    case 'file':
-                      final String? result = await FlutterFileDialog.pickFile();
-                      if (result != null) {
-                        controller.text = result;
-                        widget.onChanged(controller.text);
+                  // manageExternalStorage permission is required for file/folder selection
+                  // otherwise, the app will not complain, but the patches will error out
+                  // the same way as if the user selected an empty file/folder.
+                  // Android 11 and above requires the manageExternalStorage permission
+                  final Map<String, dynamic> availableActions = {
```

Comment:
Reduce code duplication
```suggestion
                  if (selection == 'file' || selection == 'folder') {
                    if (_managerAPI.isScopedStorageAvailable) {
                      final permission =
                          await Permission.manageExternalStorage.request();
                      if (!permission.isGranted) {
                        return;
                      }
                    }
                  }
```